### ISSUE TITLE
SecurityGroupRule.ec2 stuck in an update loop due to default value

### ISSUE THREAD

**Issue by mbbush** [crossplane-contrib/provider-upjet-aws#1073]
### What happened?

After upgrading from provider-aws 0.43 to 0.47.1, I noticed that all my SecurityGroupRule references had an AsyncUpdateFailed condition, that they were continuously retrying.

```
  - lastTransitionTime: "2023-12-16T04:22:45Z"
    message: 'async update failed: refuse to update the external resource: cannot
      change the value of the argument "self" from "" to "false"'
    reason: AsyncUpdateFailure
    status: "False"
    type: LastAsyncOperation
```
Both `spec.forProvider.self` and `status.atProvider.self` are null in the yaml manifest (these security group rules specify `cidrBlocks`)

This seems like it's probably caused by https://github.com/upbound/provider-aws/blob/5d2894e4b9ec0cd7984aef60b3db6aa519ed84fa/config/ec2/config.go#L219-L225.  

I don't know what @ulucinar was trying to fix when he put that in there, although I saw it in a few other resources as well.

Conceptually, given the AWS/Terraform documentation that states `cidrBlocks`, `ipv6CidrBlocks`, `self`, and `sourceSecurityGroupId` are mutually exclusive, it makes sense to me that all but one of those properties should be null, and not a defined, zero value, hence my confusion about why it's being set to false. 

### How can we reproduce it?

Use the example manifest in https://github.com/upbound/provider-aws/blob/cfbafecf72eee7453517222fe7f2631fa057f733/examples/ec2/securitygrouprule.yaml (from PR #1072)

Install a cli-forking provider. Apply the manifest. Upgrade to the latest no-fork provider. Observe the issue.

Alternatively, you can artificially reproduce the issue by applying the manifest to the latest provider, then editing the `status` subresource to remove `status.atProvider.self: false` (which is the state that earlier versions of the provider left it in), and deleting the provider pod.

I can apply the same workaround in reverse (updating the status of my existing resources to have `status.atProvider.self: false`, and that removes the error, but it doesn't seem like the "right" solution, and I'd really like to learn more about Alper's thinking around the null/zero values and defaulting.

### What environment did it happen in?

* Crossplane Version: 1.12.3
* Provider Version: Upgrade from 0.43.0 to 0.47.1
* Kubernetes Version:  
* Kubernetes Distribution:

**Comment by haarchri** [comment]
i wonder if it's the same for https://marketplace.upbound.io/providers/upbound/provider-aws-ec2/v0.47.1/resources/ec2.aws.upbound.io/SecurityGroupIngressRule/v1beta1 and https://marketplace.upbound.io/providers/upbound/provider-aws-ec2/v0.47.1/resources/ec2.aws.upbound.io/SecurityGroupEgressRule/v1beta1 ( the plus here is the external-name sgr-xxxxxx)

if these resources working i vote for deprecation of  https://marketplace.upbound.io/providers/upbound/provider-aws-ec2/v0.47.1/resources/ec2.aws.upbound.io/SecurityGroupRule/v1beta1

**Comment by turkenf** [comment]
Hi @mbbush,

Unless I'm missing something, I couldn't reproduce the issue with the steps below.

- Installed `upbound/provider-aws-ec2:v0.43.0`
```
> k get providers.pkg.crossplane.io
NAME                          INSTALLED   HEALTHY   PACKAGE                                               AGE
provider-aws-ec2              True        True      xpkg.upbound.io/upbound/provider-aws-ec2:v0.43.0      28m
upbound-provider-family-aws   True        True      xpkg.upbound.io/upbound/provider-family-aws:v0.47.1   28m
```
- Applied this [YAML](https://github.com/upbound/provider-aws/blob/cfbafecf72eee7453517222fe7f2631fa057f733/examples/ec2/securitygrouprule.yaml), and waited for the resources to be created successfully.
- Upgraded to the `upbound/provider-aws-ec2:v0.47.1`
```
> k get providers.pkg.crossplane.io -w
NAME                          INSTALLED   HEALTHY   PACKAGE                                               AGE
provider-aws-ec2              True        True      xpkg.upbound.io/upbound/provider-aws-ec2:v0.47.1      32m
upbound-provider-family-aws   True        True      xpkg.upbound.io/upbound/provider-family-aws:v0.47.1   32m
```
- I waited for a sufficient time and added a new label to the `SecurityGroupRule` resource.
- I did not observe any errors and the resources were deleted successfully
```
> kubectl delete managed  --all
securitygroup.ec2.aws.upbound.io "securitygrouprule" deleted
vpc.ec2.aws.upbound.io "securitygrouprule" deleted
securitygrouprule.ec2.aws.upbound.io "securitygrouprule" deleted
 ~
```

**Comment by mbbush** [comment]
I'm puzzled by this one. 

I can consistently reproduce it by editing the status.atProvider.self to be null while the provider pod scaled to zero. In my production cluster, I've observed this on all 4 out of 4 securitygrouprule resources, while in our test clusters I've observed it on about 30-40% of the managed resources. I don't see any particular pattern, and often multiple rules managed by the same composite will show different behavior.
